swupd/manifest.go

```diff
@@ -531,3 +531,38 @@ func getManifestVerFromMoM(mom *Manifest, b *Manifest) uint32 {
 
 	return 0
 }
+
+// maxFullFromManifest maximizes all file entries in mf that are also in bundle
+func maxFullFromManifest(mf *Manifest, bundle *Manifest) {
+	mf.sortFilesName()
```

Comment:
Should maxFullFromManifest sort the files in the manifests as a side effect, or should it make a copy of the array of file pointers and sort that copy? Obviously we want the mf.Files[x].Version fields updated, but not clear we want the order of the mf.Files and bundle.Files altered.

 Should maxFullFromMinifest be a method on mf?

Reply:
> Should maxFullFromManifest sort the files in the manifests as a side effect
If you look in #82 I have a reason for wanting to actually sort the manifests. This may or may not be a good idea because of side effects, but it allows us to skip sorting if we walk the manifest several times.

> Should maxFullFromManifest be a method on mf?
I avoided that because it is very specifically meant to be used on the Manifest.full and no other manifest.

Reply:
Fair enough. Just noting that #82 didn't exist when I wrote my review, and it doesn't prevent there being a separate sort step. I think that a few extra comments explaining the reasoning would help. The array being sorted makes concurrent access to the manifest more fragile, even if it is only read access.

OK also on the reasoning behind not making it a method.